webpack/components/AnsibleHostDetail/components/AnsibleVariableOverrides/AnsibleVariableOverridesHelper.js

```diff
@@ -0,0 +1,8 @@
+export const reorderVariables = variables =>
+  variables.reduce((memo, role) => {
+    const vars = role?.ansibleVariablesWithOverrides?.nodes.map(variable => ({
+      ...variable,
+      roleName: role.name,
+    }));
+    return memo.concat(vars);
+  }, []);
```

Comment:
Can we get the data from the server in the format we want it directly and avoid this complex mutation? IIUC all this changes is adding a roleName field on every variable, but I may be misunderstanding?